src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -56,10 +85,53 @@ class CollapseTuple(eve.PreserveLocationVisitor, eve.NodeTranslator):
       - `tuple_get(i, make_tuple(e_0, e_1, ..., e_i, ..., e_N))` -> `e_i`
     """
 
+    # TODO(tehrengruber): This Flag machanism is a little low level. What we actually want
+    #   is something like a pass manager, where for each pattern we have a corresponding
+    #   transformation, etc.
+    class Flag(enum.IntEnum):
+        #: `make_tuple(tuple_get(0, t), tuple_get(1, t), ..., tuple_get(N-1,t))` -> `t`
+        COLLAPSE_MAKE_TUPLE_TUPLE_GET = 1
+        #: `tuple_get(i, make_tuple(e_0, e_1, ..., e_i, ..., e_N))` -> `e_i`
+        COLLAPSE_TUPLE_GET_MAKE_TUPLE = 2
+        #: `tuple_get(i, let(...)(make_tuple()))` -> `let(...)(tuple_get(i, make_tuple()))`
+        PROPAGATE_TUPLE_GET = 4
+        #: `{1, 2}` -> `(λ(_tuple_el_1, _tuple_el_2) → {_tuple_el_1, _tuple_el_2})(1, 2)`
+        LETIFY_MAKE_TUPLE_ELEMENTS = 8
+        #: Inverse of LETIFY_MAKE_TUPLE_ELEMENTS run after all other transformations
+        #: `(λ(_tuple_el_1, _tuple_el_2) → {_tuple_el_1, _tuple_el_2})(1, 2)` -> {1, 2}`
+        REMOVE_LETIFIED_MAKE_TUPLE_ELEMENTS = 16
+        #: `let(tup, {trivial_expr1, trivial_expr2})(foo(tup))`
+        #:  -> `foo({trivial_expr1, trivial_expr2})`
+        INLINE_TRIVIAL_MAKE_TUPLE = 32
+        #: `(if cond then {1, 2} else {3, 4})[0]` -> `if cond then {1, 2}[0] else {3, 4}[0]`
+        PROPAGATE_TO_IF_ON_TUPLES = 64
+        #: `let((a, let(b, 1)(a_val)))(a)`-> `let(b, 1)(let(a, a_val)(a))`
+        PROPAGATE_NESTED_LET = 128
+        #: `let(a, 1)(a)` -> `1`
+        INLINE_TRIVIAL_LET = 256
+
     ignore_tuple_size: bool
-    collapse_make_tuple_tuple_get: bool
-    collapse_tuple_get_make_tuple: bool
     use_global_type_inference: bool
+    flags: int = (
```

Comment:
I don't like this way of expression "all". Maybe with enum.Flag there is a better way, see https://stackoverflow.com/questions/42251081/representation-of-all-values-in-flag-enum